mlrun/runtimes/nuclio/application/application.py

```diff
@@ -527,21 +562,22 @@ def _build_application_image(
             with_mlrun=with_mlrun,
         )
 
-    def _ensure_reverse_proxy_configurations(self):
-        if self.spec.build.functionSourceCode or self.status.container_image:
+    @staticmethod
+    def _ensure_reverse_proxy_configurations(function):
+        if function.spec.build.functionSourceCode or function.status.container_image:
```

Comment:
What is the reason of making this method static if it changes the `self`?

Reply:
I wanted to reuse this in the new class method `deploy_reverse_proxy_image`, where we actually create a `RemoteRuntime` instance. to ensure this is done on this instance, I changed the function to get a function instance as an argument, then it can be a non-application runtime function.

Reply:
@TomerShor I see. `self._ensure_reverse_proxy_configurations(self)` looks a bit weird, so maybe worth adding a comment explaining the trick. Also, please add type hint for a `function` arg